pkg/lifecycle/version_rules.go

```diff
@@ -0,0 +1,195 @@
+	errorBranchVersionNotInRules error = errors.New("the given branch version is not defined in the rules")
+)
+
+// Version holds the maximum and minimum limits allowed for a specific branch version
+type Version struct {
+	Min string `json:"min"`
+	Max string `json:"max"`
+}
+
+// VersionRules will hold all the necessary information to check which assets versions are allowed to be in the repository
+type VersionRules struct {
+	Rules            map[string]Version `json:"rules"`
+	BranchVersion    string             `json:"branch-version,omitempty"`
+	MinVersion       int                `json:"min-version,omitempty"`
+	MaxVersion       int                `json:"max-version,omitempty"`
+	DevBranch        string             `json:"dev-branch,omitempty"`
+	DevBranchPrefix  string             `json:"dev-branch-prefix"`
+	ProdBranch       string             `json:"prod-branch,omitempty"`
+	ProdBranchPrefix string             `json:"prod-branch-prefix"`
+}
+
+// loadFromJSON will load the version rules from version_rules.json file at charts repository
+func loadFromJSON(fs billy.Filesystem) (*VersionRules, error) {
+	vr := &VersionRules{
+		Rules: make(map[string]Version),
+	}
+
+	if exist, err := filesystem.PathExists(fs, path.VersionRulesFile); err != nil || !exist {
+		return nil, err
+	}
+
+	file, err := os.Open(filesystem.GetAbsPath(fs, path.VersionRulesFile))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	if err := json.NewDecoder(file).Decode(&vr); err != nil {
+		return nil, err
+	}
+
+	return vr, nil
+}
+
+type jsonLoader func(fs billy.Filesystem) (*VersionRules, error)
+
+// rules will check and convert the provided branch version,
+// create the hard-coded rules for the charts repository and calculate the minimum and maximum versions according to the branch version.
+func (d *Dependencies) rules(branchVersion string, jsonLoad jsonLoader) (*VersionRules, error) {
+	if branchVersion == "" {
+		return nil, errorNoBranchVersion
+	}
+
+	v, err := jsonLoad(d.RootFs)
+	if err != nil {
+		return v, err
+	}
+
+	if _, ok := v.Rules[branchVersion]; !ok {
+		return nil, errorBranchVersionNotInRules
+	}
+
+	v.BranchVersion = branchVersion
+
+	// Calculate the min and maximum versions allowed for the current branch version lifecycle
+	if err := v.getMinMaxVersionInts(); err != nil {
+		return nil, err
+	}
+
+	v.ProdBranch = v.ProdBranchPrefix + v.BranchVersion
+	v.DevBranch = v.DevBranchPrefix + v.BranchVersion
+
+	return v, nil
+}
+
+// Current lifecycle rules are:
+//
+//	Branch can only hold until 2 previous versions of the current branch version.
+//	Branch cannot hold versions from newer branches, only older ones.
+//
+// See CheckChartVersionForLifecycle() for more details.
+func (v *VersionRules) getMinMaxVersionInts() error {
+	// e.g: 2.9 - 0.2 = 2.7
+	minVersionStr := v.Rules[(branchVersionMinorSum(v.BranchVersion, -2))].Min
+	maxVersionStr := v.Rules[v.BranchVersion].Max
+
+	var err error
```

Comment:
Error doesn't need and shouldn't be predeclared.

Reply:
In this case I need it, otherwise I will get an error saying that `min` and `max` variable are declared and not used. 
@briandowns